MDNetworking/MemberImpl/MDReplicatedMember.cs

```diff
@@ -0,0 +1,165 @@
+using Godot;
+using System;
+using System.Reflection;
+
+public class MDReplicatedMember
+{
+    public bool ProcessWhilePaused { get; set; } = true;
+
+    public string ReplicationGroup { get; set; } = null;
+
+    protected const string LOG_CAT = "LogReplicatedMember";
+
+    protected MemberInfo Member;
+
+    protected object LastValue;
+
+    protected bool Reliable;
+
+    protected MDReplicatedType ReplicatedType;
+
+    protected WeakRef NodeRef;
+
+    protected bool IsShouldReplicate = false;
+
+    public MDReplicatedMember(MemberInfo Member, bool Reliable, MDReplicatedType ReplicatedType, WeakRef NodeRef, MDReplicatedSetting[] Settings)
+    {
+        MDLog.AddLogCategoryProperties(LOG_CAT, new MDLogProperties(MDLogLevel.Info));
+        this.Member = Member;
+        this.Reliable = Reliable;
+        this.ReplicatedType = ReplicatedType;
+        this.NodeRef = NodeRef;
+        CheckIfShouldReplicate();
+    }
+
+    ///<summary>Proviedes a unique key to identify this member</summary>
+    public String GetUniqueKey()
+    {
+        Node Node = NodeRef.GetRef() as Node;
+        return Node.GetPath() + "#" + Member.Name;
+    }
+
+    protected virtual object GetValue()
+    {
+        Node Instance = NodeRef.GetRef() as Node;
+        FieldInfo Field = Member as FieldInfo;
+
+        if (Field != null)
+        {
+            return Field.GetValue(Instance);
+        }
+
+        PropertyInfo Property = Member as PropertyInfo;
+        if (Property != null)
+        {
+            return Property.GetValue(Instance);
+        }
+
+        return null;
+    }
+
+    ///<summary>It is better to implement ReplicateToAll and ReplicateToPeer instead of this method</summary>
+    public virtual void Replicate(int JoinInProgressPeerId, bool IsIntervalReplicationTime)
+    {
+        object CurrentValue = GetValue();
+        Node Instance = NodeRef.GetRef() as Node;
+
+        if ((ReplicatedType == MDReplicatedType.Interval && IsIntervalReplicationTime) || 
+        (object.Equals(LastValue, CurrentValue) == false && ReplicatedType == MDReplicatedType.OnChange))
```

Comment:
Should do the OnChange check before the object.Equals as it's slower

Reply:
Fixed